program/tests/namespace_royalty_enforcement.rs

```diff
@@ -0,0 +1,1123 @@
+        )
+        .unwrap();
+    royalty_rule_set
+        .add(
+            token_delegate_transfer_operation.to_string(),
+            rules.namespace_rule.clone(),
+        )
+        .unwrap();
+
+    // --------------------------------
+    // NOTE THIS IS THE ONLY OPERATION
+    // THAT USES THE ADVANCED DELEGATE
+    // RULE.
+    // --------------------------------
+    royalty_rule_set
+        .add(
+            token_delegate_locked_transfer_operation.to_string(),
+            rules.advanced_delegate_rule,
+        )
+        .unwrap();
+
+    royalty_rule_set
+        .add(
+            token_delegate_utility_operation.to_string(),
+            rules.namespace_rule.clone(),
+        )
+        .unwrap();
+
+    royalty_rule_set
+        .add(
+            token_delegate_staking_operation.to_string(),
+            rules.namespace_rule,
+        )
+        .unwrap();
+
+    print!("Royalty Rule Set: {:#?}", royalty_rule_set);
+
+    royalty_rule_set
+}
+
+async fn create_royalty_rule_set(context: &mut ProgramTestContext) -> Pubkey {
+    let royalty_rule_set = get_royalty_rule_set(context.payer.pubkey());
+
+    // Put the `RuleSet` on chain.
+    create_big_rule_set_on_chain!(
+        context,
+        royalty_rule_set.clone(),
+        RULE_SET_NAME.to_string(),
+        Some(ADDITIONAL_COMPUTE)
+    )
+    .await
+}
+
+async fn create_incomplete_royalty_rule_set(
+    context: &mut ProgramTestContext,
+    missing_op: String,
+) -> Pubkey {
+    let mut royalty_rule_set = get_royalty_rule_set(context.payer.pubkey());
+    // Remove a namespaced operation to verify it fails.
+    royalty_rule_set.operations.remove(&missing_op);
+
+    // Put the `RuleSet` on chain.
+    create_big_rule_set_on_chain!(
+        context,
+        royalty_rule_set.clone(),
+        RULE_SET_NAME.to_string(),
+        Some(ADDITIONAL_COMPUTE)
+    )
+    .await
+}
+
+#[tokio::test]
+async fn create_rule_set() {
+    let mut context = program_test().start_with_context().await;
+    let _rule_set_addr = create_royalty_rule_set(&mut context).await;
+}
+
+#[tokio::test]
+async fn wallet_to_wallet_unimplemented() {
+    let mut context = program_test().start_with_context().await;
+    let rule_set_addr = create_royalty_rule_set(&mut context).await;
+
+    // Create a Keypair to simulate a token mint address.
+    let mint = Keypair::new();
+
+    // Create source and destination wallets.
+    let source = Keypair::new();
+    let dest = Keypair::new();
+
+    // Store the payload of data to validate against the rule definition.
+    let payload = Payload::from([
+        (PayloadKey::Amount.to_string(), PayloadType::Number(1)),
+        (
+            PayloadKey::Source.to_string(),
+            PayloadType::Pubkey(source.pubkey()),
+        ),
+        (
+            PayloadKey::Destination.to_string(),
+            PayloadType::Pubkey(dest.pubkey()),
+        ),
+    ]);
+
+    let transfer_wallet_to_wallet_operation = Operation::Transfer {
```

Comment:
assuming most of these tests are the same as what was in basic royalty enforcement test, lmk if theres any specific area to check other than what I checked at the top which was the new ruleset construction using namespace